_posts/en/newsletters/2024-08-02-newsletter.md

```diff
@@ -0,0 +1,197 @@
+  transaction belonging to a unique chunk or sub-chunk.  That makes it
+  possible to precompute each transaction's chunk or sub-chunk, called
+  its _absorption set_, and associate that with the transaction.  Wuille
+  shows how the existing chunking algorithm already calculates each
+  transaction's absorption set.
+
+  When a miner has filled a template with all of the full chunks
+  possible, it can take the precomputed absorption sets for all
+  transactions not yet included in the block and consider them in feerate
+  order.  This only requires a single sort operation on a list with the
+  same number of elements as there are transactions in the mempool
+  (almost always less than a million with current defaults).  The best
+  feerate absorption sets (chunks and sub-chunks) can then be used to
+  fill the remaining block space.  This requires tracking the number of
+  transactions from a cluster that have been included so far and
+  skipping any sub-chunks that don't fit or which have already had some
+  of their transactions included.
+
+  However, although chunks can be compared with each other to provide the best
+  order for block inclusion, the individual transactions within a chunk
+  or sub-chunk are not guaranteed to be in the best order for only
+  including some of those transactions.  That can lead to non-optimal
+  selection when a block is nearly full.  For example, when only 300
+  vbytes remain, the algorithm might select a 200-vbyte transaction at
+  5 sats/vbyte (1,000 sats total) instead of two 150-vbyte transactions
+  at 4 sats/vbyte (1,200 sats total).
+
+  Wuille describes how precomputed absorption sets are especially useful
+  in this case: because they only require tracking the number of
+  transactions from each cluster that have been included so far, they
+  make it easy to restore to an earlier state in the template-filling
+  algorithm and replace the previously made choice with an alternative
+  to see if it results in collecting more total fees.  This
+  allows implementing a [branch-and-bound][] search that can try many
+  combinations of filling the last bit of block space in the hopes of
+  finding a better result than the simple algorithm.
+
+- **Hyperion network event simulator for the Bitcoin P2P network:**
+  Sergi Delgado [posted][delgado hyperion] to Delving Bitcoin about a
+  [Hyperion][], a network simulator he's written that tracks how data
+  propagates through a simulated Bitcoin network.  The work is initially
+  motivated by a desire to compare Bitcoin's current method for relaying
+  transaction announcements (`inv` inventory messages) to the proposed
+  [Erlay][topic erlay] method.
+
+## Releases and release candidates
+
+*New releases and release candidates for popular Bitcoin infrastructure
+projects.  Please consider upgrading to new releases or helping to test
+release candidates.*
+
+- [BDK 1.0.0-beta.1][] is a release candidate for "the beta first version of
```

Comment:
```suggestion
- [BDK 1.0.0-beta.1][] is a release candidate for "the first beta version of
```
Also, I don't think this is even a "release candidate", it's certainly not called that in the release notes.

Reply:
Ping BDK maintainer @notmandatory

Interesting.  I may have been misled by GitHub's UI calling it a "Pre-Release":

![2024-07-31-12_04_43_356780437](https://github.com/user-attachments/assets/ce36a3f0-d17f-4881-90bb-afb5469d59b9)

Comparing to LND, which uses `-beta` for its main releases, it looks like "Pre-Release" is set for their RCs but not their main releases:

![2024-07-31-12_08_56_620346134](https://github.com/user-attachments/assets/e7aef132-a3db-406f-b3b6-d72c757df1cd)

![latest](https://github.com/user-attachments/assets/e177926a-f905-4bba-ad1c-fa61d6c31197)

I suggest I make the transposition fix now but that we leave it described as an RC for this newsletter, see what @notmandatory says, and make any required changes to our announcement for next week's newsletter.

Reply:
Hey sorry I just saw this in my notifications. BDK 1.0-beta.1 is the first release candidate for an upcoming final BDK 1.0 release. It's also worth noting that the original `bdk` rust crate has been renamed to `bdk_wallet` (https://crates.io/crates/bdk_wallet).

The old `bdk` crate used to include everything but was renamed `bdk_wallet` with lower layer modules extracted into their own crates (`bdk_chain`, `bdk_electrum`, `bdk_esplora`, `bdk_bitcoind_rpc`). Most users will only see `bdk_wallet` and which ever chain syncing crate(s) they want to use.

The purpose of the 1.0-beta.x releases is to:

1. help users update their apps to a ~stable 1.0 `bdk_wallet` API
2. improve testing and API docs
3. update and add examples and tutorials
4. update language bindings (kotlin, swift, python)

Hope this all makes sense, if not just let me know.